src/AutoRest.CSharp.V3/ClientModels/ClientBuilder.cs

```diff
@@ -191,10 +197,63 @@ private static ServiceClientParameter[] OrderParameters(IEnumerable<ServiceClien
                 request,
                 OrderParameters(methodParameters),
                 clientResponse,
-                new ClientMethodDiagnostics($"{clientName}.{operationName}", Array.Empty<DiagnosticScopeAttributes>())
+                new ClientMethodDiagnostics($"{clientName}.{operationName}",Array.Empty<DiagnosticScopeAttributes>())
             );
         }
 
+        private static bool IsNextPageParameter(ServiceClientParameter parameter) =>
+            parameter.Location != ParameterLocation.Path
+            && parameter.Location != ParameterLocation.Body
+            && parameter.Location != ParameterLocation.Query;
+
+        private static ClientMethod BuildNextPageMethod(ClientMethod method)
+        {
+            var nextPageUrlParameter = new ServiceClientParameter(
+                "nextLinkUrl",
+                "The URL to the next page of results.",
+                new FrameworkTypeReference(typeof(string)),
+                null,
+                true,
+                ParameterLocation.Uri);
+            var parameters = method.Parameters.Where(IsNextPageParameter).Append(nextPageUrlParameter).ToArray();
```

Comment:
Why do we need to filter original parameters here?

Reply:
Because NextPage doesn't take Path, Body, or Query parameters. That's the only reason we can use the WriteOperation logic as is, because we filter those here.

Reply:
What about just adding a single next page parameter?

Reply:
I'm not sure if we need to be sending the original headers with GetNext call.

Reply:
I originally only had only nextPageUrl with it. But, I realized it needed all parameters that are not related to the Uri itself. The headers are the only extra parameters in the paging tests on the test server, which include maxresults, which was what the pageSizeHint should be. We can ask today at the autorest meeting if services ever need headers when following nextLink urls. But, it doesn't hurt to have it send them sent for now. So, as an initial implementation of paging, this is fine. If we find we can remove it because it isn't supported, we'll remove it.

Reply:
I think my biggest issue is that I don't want more code taking dependency on location property of parameter. So if the feature is required I would rather design it differently.

Reply:
How about, when the m4 changes for paging information are implemented, we revisit what is necessary and cut back/redesign? That way, we can have a working paging mechanism for something like AppConfiguration, and make the adjustments necessary to reduce it when we have the 'final' input from m4. Sound good?

Reply:
I don't understand why do you want to keep features that we don't need? Do we need headers for app configuration to work?

Reply:
Ok, since we decided to send headers let's keep copying header paramters but still remove Location from parameter definition. We can figure out which parameters need copying just by looking at `method.Request.Headers`

Reply:
Got it working. Fixed.